### ISSUE TITLE
Interpolation creates null values in resampled dataframe

### ISSUE THREAD

**Issue by lambertsbennett** [databrickslabs/tempo#320]
I am having an issue with resampling/interpolation that I think must be a simple misunderstanding. I have a signal at 10 Hz and I want to upsample to 100 Hz and fill values with a linear interpolation. My current code is as follows:

```python
interp_df = start_tsdf.resample(freq=f"{1/100} sec", func='mean').interpolate(target_cols=['value'], method='linear', show_interpolated=True)
```

However this results in a value column that contains only nulls in between actual sensor readings. Is it not possible to upsample a signal and linearly interpolate to fill missing values?

**Comment by tnixon** [comment]
You should be able to do this @lambertsbennett, if not this is definitely a bug. Could you provide some sample data we could test with?

**Comment by lambertsbennett** [comment]
@tnixon thanks for the response, I will take a small sample of the data when I'm next at work and provide it as soon as possible!

**Comment by lambertsbennett** [comment]
[sample.csv](https://github.com/databrickslabs/tempo/files/11372956/sample.csv)
Hi here are some sample data that I can use to reproduce the error. In this case I am using the signal and the file_name columns as partition columns. What I am thinking could be the problem is that many of the values are identical and it could be that this is a strange case?

**Comment by tnixon** [comment]
Thanks @lambertsbennett - I will investigate ASAP!

**Comment by lambertsbennett** [comment]
@tnixon I have also been looking more into this and things are even a bit stranger... with a different signal in the data there is a large time gap and this large time gap is interpolated properly, but the smaller gaps are filled with nulls.

**Comment by lambertsbennett** [comment]
Hi @tnixon, I was just wondering if there are any updates on this issue?

Thanks!

**Comment by lambertsbennett** [comment]
@tnixon we eventually completely switched up our interpolation process, but it would be nice in the future to use tempo. Did anything ever come out of this?

**Comment by tnixon** [comment]
Hi @lambertsbennett, thanks for providing the sample data. I'm sorry we haven't been able to follow up on this yet, but it is still in the queue and valuable to us to figure out what is going on here. I am setting aside some time to look at it in the near term

**Comment by jplavins** [comment]
Hello! I have the same problem where I end up with null values as a result of resampling/interpolation.
I've attached the sample data. Some combination of parameters work, some return null values (func='ceil' and method='linear').
[data_sample.csv](https://github.com/user-attachments/files/16813648/data_sample.csv)

**Comment by tnixon** [comment]
Thanks for letting us know @jplavins - we are currently all focused on a major update to the Tempo API, v0.2 which should be released in the new year. We will put in test code to make sure this issue does not come up in the new version.

**Comment by R7L208** [comment]
## Triage — March 2026

**Status**: Likely fixed on `v0.2-integration`. Interpolation was completely rewritten in v0.2 using pandas UDFs with per-partition processing. The old approach used Spark window functions that could produce null artifacts at partition boundaries — the new approach processes each series independently in pandas, which should eliminate boundary-related null values.

**Jobs to be done**: Create a repro test matching the original issue's scenario (interpolation producing unexpected nulls) and verify it no longer occurs on v0.2.

**Effort estimate**: 🟡 Small–Medium (1-2 days) — need to reproduce the original scenario

**Closure path**: Close after v0.2 merge + repro test confirms nulls are no longer created.
